### PR TITLE
fix: add missing component imports to TS demos

### PR DESCRIPTION
**Pull request by web-padawan**
## Description

Added missing imports to TS demos (currently some of them are broken when opening in a new window).

### CODE REVIEW

File: frontend/demo/component/checkbox/checkbox-adjacent-groups.ts [hunk 1]
```diff
@@ -3,6 +3,8 @@ import 'Frontend/demo/init'; // hidden-source-line
 import { html, LitElement } from 'lit';
 import { customElement } from 'lit/decorators.js';
 import '@vaadin/checkbox';
+import '@vaadin/checkbox-group';
+import '@vaadin/vertical-layout';
```
Comment (web-padawan) [inline frontend/demo/component/checkbox/checkbox-adjacent-groups.ts:7]: This [demo](https://vaadin.com/docs/latest/example?path=component/checkbox/checkbox-adjacent-groups.ts) is currently broken when opening in a new tab.

File: frontend/demo/component/checkbox/checkbox-custom-presentation.ts [hunk 2]
```diff
@@ -3,6 +3,7 @@ import 'Frontend/demo/init'; // hidden-source-line
 import { html, LitElement } from 'lit';
 import { customElement, state } from 'lit/decorators.js';
 import '@vaadin/checkbox';
+import '@vaadin/checkbox-group';
```
Comment (web-padawan) [inline frontend/demo/component/checkbox/checkbox-custom-presentation.ts:6]: This demo is not currently shown on the website, was hidden in https://github.com/vaadin/docs/commit/1a9be5c250dbddf26bc46914819932602070f1d4

File: frontend/demo/component/checkbox/checkbox-horizontal.ts [hunk 3]
```diff
@@ -3,6 +3,7 @@ import 'Frontend/demo/init'; // hidden-source-line
 import { html, LitElement } from 'lit';
 import { customElement } from 'lit/decorators.js';
 import '@vaadin/checkbox';
+import '@vaadin/checkbox-group';
```
Comment (web-padawan) [inline frontend/demo/component/checkbox/checkbox-horizontal.ts:6]: This [demo](https://vaadin.com/docs/latest/example?path=component/checkbox/checkbox-horizontal.ts) is currently broken when opening in a new tab.

File: frontend/demo/component/confirmdialog/confirm-dialog-cancel-button.ts [hunk 4]
```diff
@@ -1,7 +1,9 @@
 import 'Frontend/demo/init'; // hidden-source-line
 import { html, LitElement } from 'lit';
 import { customElement, state } from 'lit/decorators.js';
+import '@vaadin/button';
```
Comment (web-padawan) [inline frontend/demo/component/confirmdialog/confirm-dialog-cancel-button.ts:4]: While `@vaadin/button` is imported by the `@vaadin/confirm-dialog`, it still makes sense to import it here.

File: frontend/demo/component/confirmdialog/confirm-dialog-cancel-button.ts [hunk 5]
```diff
@@ -1,7 +1,9 @@
...
 import '@vaadin/confirm-dialog';
+import '@vaadin/horizontal-layout';
```
Comment (web-padawan) [inline frontend/demo/component/confirmdialog/confirm-dialog-cancel-button.ts:6]: This [demo](https://vaadin.com/docs/latest/example?path=component/confirmdialog/confirm-dialog-cancel-button.ts) does not have the `vaadin-horizontal-layout` defined when opening in a new tab.

File: frontend/demo/component/grid/grid-column-header-footer.ts [hunk 6]
```diff
@@ -9,6 +9,7 @@ import {
   columnHeaderRenderer,
   GridColumnBodyLitRenderer,
 } from '@vaadin/grid/lit.js';
+import '@vaadin/horizontal-layout';
```
Comment (web-padawan) [inline frontend/demo/component/grid/grid-column-header-footer.ts:12]: This [demo](https://vaadin.com/docs/latest/example?path=component/grid/grid-column-header-footer.ts) does not have `vaadin-horizontal-layout` defined when opening in a new tab.

**Review by tomivirkki — APPROVED** [review]

**Pull request merged**